### DISCUSSION TITLE
What JSON key-value input will pass validation if my schema specifies oneOf ("type" : "null" and "type" : "number") ?

### DISCUSSION THREAD

**Discussion by GaryRoussak** [json-schema-org/community#592] [Q&A]
Hi,

I've picked up an old buggy project where { "port" : "NULL" } is being failed by a JSON-Schema validator that expects the port key to have a number or be null. I completely understand why this is failing. "NULL" is not a number, and it is not null ! I realise therefore that the code generating the JSON in this case is at fault.

I don't want to change the validator. Furthermore, the "port" value is actually null and needs to stay that way. But how do I represent the port value in JSON for it to pass validation ?

Should it be:
{ "port" : "null" }
or
{ "port" : null }

Thanks

**Comment by Relequestual** [comment]
Hello. Thanks for your question. Can you include the schema in your question as opposed to the title pleaes? =]

**Comment by Relequestual** [comment]
It should be
`{"port": null}`

You can see this working here: https://jsonschema.dev/s/56J4F

**Comment by GaryRoussak** [comment]
Thanks Ben, that’s perfect – all I needed.

Regards
Gary Roussak
Senior Software Engineer
Forest Rock Systems Ltd  |   Charnwood Building  |  Holywell Park  |  Loughborough University  |  Ashby Road   |
Loughborough |  Leicestershire  |  LE11 3AQ

T: 0330 660 0567  |  M: 07507 800 308 | W: www.forestrock.co.uk<http://www.forestrock.co.uk/>  |  E: ***@***.******@***.***>

***@***.***
